src/ucs/sys/netlink.c

```diff
@@ -249,44 +248,63 @@ ucs_netlink_parse_rt_entry_cb(const struct nlmsghdr *nlh, void *arg)
     return UCS_INPROGRESS;
 }
 
+static void
+ucs_netlink_lookup_in_iface_rules(ucs_netlink_route_info_t *info,
+                                  ucs_netlink_rt_rules_t *iface_rules)
+{
+    ucs_netlink_route_entry_t *curr_entry;
+
+    ucs_array_for_each(curr_entry, iface_rules) {
+        if (curr_entry->subnet_prefix_len > info->found_netmask_len) {
+            if (ucs_sockaddr_is_same_subnet(
+                    info->sa_remote, (const struct sockaddr *)&curr_entry->dest,
+                    curr_entry->subnet_prefix_len)) {
```

Comment:
1. can unite the 2 if conditions
2. can unite "found" and "found_netmask_len" fields: before found, can set found_netmask_len to -1 (assuming this is signed integer type field)